crates/admin/src/cluster_controller/service.rs

```diff
@@ -650,14 +692,18 @@ where
         &mut self,
         node_id: GenerationalNodeId,
         partition_id: PartitionId,
+        min_target_lsn: Option<Lsn>,
     ) -> anyhow::Result<Snapshot> {
         // todo(pavel): make snapshot RPC timeout configurable, especially if this includes remote upload in the future
```

Comment:
It might be nice to take an external deadline - regardless of this timeout, the server won't give up (unless it crashes or is stopped) driving the snapshot to completion.

Reply:
You can cascade down the grpc deadline if it the snapshot request originated from a grpc call